src/lib/tls/tls12/msg_client_kex.cpp

```diff
@@ -114,19 +121,10 @@ Client_Key_Exchange::Client_Key_Exchange(Handshake_IO& io,
             append_tls_length_value(m_pre_master, psk.bits_of(), 2);
          }
 
-         if(curve_id.is_ecdh_named_curve()) {
-            auto* ecdh_key = dynamic_cast<ECDH_PublicKey*>(private_key.get());
```

Comment:
This line caused the concrete issue: `private_key` is passed in by the user and must derive from `PK_Key_Agreement_Key`. But if it handles a known ECDH group we just try to downcast to `ECDH_PublicKey` to handle the `EC_Point_Format` selection (compressed vs. uncompressed) in the serialization of the public value.